### PR TITLE
Added the integration test for blocking calls in call mediator

### PR DESCRIPTION
**Pull request by RKathees**

### CODE REVIEW

File: modules/integration/tests-integration/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/call/CallMediatorBlockingDirectEndpointTestCase.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2005-2010, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
```
Comment (vanjikumaran) [inline modules/integration/tests-integration/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/call/CallMediatorBlockingDirectEndpointTestCase.java]: Please change the year to 2015

File: modules/integration/tests-integration/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/call/CallMediatorBlockingDirectEndpointTestCase.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+import static org.testng.Assert.assertTrue;
+
+public class CallMediatorBlockingDirectEndpointTestCase extends ESBIntegrationTest {
+
+    @BeforeClass(alwaysRun = true)
+    public void setEnvironment() throws Exception {
+        super.init();
+        loadESBConfigurationFromClasspath("/artifacts/ESB/mediatorconfig/call/CallMediatorBlockingDirectEndpointTest.xml");
+    }
```
Comment (vanjikumaran) [inline modules/integration/tests-integration/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/call/CallMediatorBlockingDirectEndpointTestCase.java]: Use File.separator

File: modules/integration/tests-integration/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/call/CallMediatorBlockingDirectEndpointTestCase.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+    @Test(groups = {"wso2.esb"})
+    public void callMediatorBlockingDirectEndpointTest() throws AxisFault {
```
Comment (vanjikumaran) [inline modules/integration/tests-integration/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/call/CallMediatorBlockingDirectEndpointTestCase.java]: Give some meaning full description 

description = "XXXXXXXXXXXXXXXXXXXXXXXXX")

File: modules/integration/tests-integration/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/call/CallMediatorBlockingDirectEndpointTestCase.java (hunk 2 above)
Comment (RKathees) [inline modules/integration/tests-integration/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/call/CallMediatorBlockingDirectEndpointTestCase.java]: Used the File.separator - https://github.com/wso2/product-esb/pull/187

Comment (RKathees) [inline modules/integration/tests-integration/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/call/CallMediatorBlockingDirectEndpointTestCase.java]: Used the File.separator - https://github.com/wso2/product-esb/pull/241

File: modules/integration/tests-integration/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/call/CallMediatorBlockingDirectEndpointTestCase.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+import org.apache.axiom.om.OMElement;
+import org.apache.axis2.AxisFault;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.wso2.esb.integration.common.utils.ESBIntegrationTest;
+
+import static org.testng.Assert.assertTrue;
+
```
Comment (lafernando) [inline modules/integration/tests-integration/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/call/CallMediatorBlockingDirectEndpointTestCase.java]: Add class level comments, do it for all.

File: modules/integration/tests-integration/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/call/CallMediatorBlockingSwitchTestCase.java [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+    @BeforeClass(alwaysRun = true)
+    public void setEnvironment() throws Exception {
+        super.init();
+        loadESBConfigurationFromClasspath("/artifacts/ESB/mediatorconfig/call/CallMediatorBlockingSwitchTest.xml");
+    }
+
+    @Test(groups = {"wso2.esb"},description = "Call the endpoint with blocking external calls according to the switch case")
+    public void callMediatorBlockingSwitchTest() throws AxisFault {
+
```
Comment (lafernando) [inline modules/integration/tests-integration/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/call/CallMediatorBlockingSwitchTestCase.java]: Remove unneeded whitespace, be consistent with the coding style.

File: modules/integration/tests-integration/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/call/CallMediatorBlockingDirectEndpointTestCase.java (hunk 4 above)
Comment (katheesR) [inline modules/integration/tests-integration/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/call/CallMediatorBlockingDirectEndpointTestCase.java]: I have fixed in the PR - https://github.com/wso2/product-esb/pull/346

File: modules/integration/tests-integration/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/call/CallMediatorBlockingSwitchTestCase.java (hunk 5 above)
Comment (katheesR) [inline modules/integration/tests-integration/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/call/CallMediatorBlockingSwitchTestCase.java]: I have fixed in the PR - https://github.com/wso2/product-esb/pull/346

**Pull request merged**